### DISCUSSION TITLE
Ignore a folder only on top level of a repository

### DISCUSSION THREAD

**Discussion by reburg67** [community/community#23600] [New to GitHub]
<p>I want to exclude a folder named <code>assets</code> on top level of my git-repository. Other folders in subdirectories having the name<code> assets/</code> should not be excluded, e.g <code>/extension/assets/*</code>)</p>
<p>The folder <code>/assets</code> was in the repo until now.</p>
<p>I created a file <code>.gitignore</code> on top level and wrote <code>/assets/*</code> into this file.</p>
<p>Now I called <code>git add .</code> and made a commit: <code>git commit -m testcommit</code> then <code>git clean -in</code> and then  <code>git clean -i</code><br>
I would have thought, that the content of the folder <code>/assets/*</code> now should get removed. But nothing happens.</p>
<p>Calling  <code>git ls-files | grep assets</code> shows me, that all  <code>/assets/*</code> directories are still present.</p>
<p>Did I get something wrong?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><a href="https://git-scm.com/docs/gitignore"><code>gitignore</code> rules</a> apply only to <em>untracked</em> files. If you want to stop tracking those files, you have to explicitly remove them from the repository. If you want to keep the files on disk (in your working tree) but remove them from the repository you can use <a href="https://git-scm.com/docs/git-rm#Documentation/git-rm.txt---cached"><code>git rm --cached</code></a>.</p>
<aside class="quote no-group" data-username="reburg67" data-post="1" data-topic="221231">
<div class="title">
<div class="quote-controls"></div>
 reburg67:</div>
<blockquote>
<p>I created a file <code>.gitignore</code> on top level and wrote <code>/assets/*</code> into this file.</p>
</blockquote>
</aside>
<p>As a side note, <code>/assets/</code> would be a better rule, it ignores the directory and everything in it. It only makes a difference if you have files starting with names starting with a dot in the directory, though.</p>

**Comment by reburg67** [comment]
<p>Thanks for the clarification.</p>
<p>The command <code>git rm -r --cached assets/*</code> did the job.</p>
